sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -971,17 +971,19 @@ def dead_letter_source(self):
 
     @property
     def message_state(self):
-        # type: () -> ServiceBusMessageState
+        # type: () -> int
         """
         Defaults to Active. Represents the message state of the message. Can be Active, Deferred.
         or Scheduled.
 
-        :rtype: ~azure.servicebus.ServiceBusMessageState
+        :rtype: int
         """
         try:
             message_state = self._raw_amqp_message.annotations.get(MESSAGE_STATE_NAME)
             if not message_state:
                 return ServiceBusMessageState.ACTIVE
+            elif message_state not in (0, 1, 2):
+                return message_state
```

Comment:
I think you could simply use:
```python
try:
    return ServiceBusMessageState(message_state)
except ValueError:
    return  message_state
```

Reply:
it'll be more like 
```
            try:
                return ServiceBusMessageState(message_state)
            except ValueError:
                if not message_state:
                    return ServiceBusMessageState.ACTIVE
                return message_state
```

Reply:
Yep, though I was just referring to the `elif`.... so:
```python
if not message_state:
    return ServiceBusMessageState.ACTIVE
try:
    return ServiceBusMessageState(message_state)
except ValueError:
    return message_state
```
But either/or. both should work.